pkg/proxy/tokenreview/tokenreview.go

```diff
@@ -23,43 +26,73 @@ var (
 
 type TokenReview struct {
 	reviewRequester clientauthv1.TokenReviewInterface
+	metrics         *metrics.Metrics
 	audiences       []string
 }
 
-func New(restConfig *rest.Config, audiences []string) (*TokenReview, error) {
+func New(restConfig *rest.Config, metrics *metrics.Metrics, audiences []string) (*TokenReview, error) {
 	kubeclient, err := kubernetes.NewForConfig(restConfig)
 	if err != nil {
 		return nil, err
 	}
 
 	return &TokenReview{
 		reviewRequester: kubeclient.AuthenticationV1().TokenReviews(),
+		metrics:         metrics,
 		audiences:       audiences,
 	}, nil
 }
 
 func (t *TokenReview) Review(req *http.Request) (bool, error) {
+	var (
+		code          int
+		user          string
+		authenticated bool
+		err           error
+	)
+
+	// Start clock on metrics
+	tokenReviewDuration := time.Now()
+	req, remoteAddr := proxycontext.RemoteAddr(req)
```

Comment:
Just to confirm.. do we want to be overwriting `req` here to include this remote address?